instrumentation/p6spy/src/test/java/brave/p6spy/ITTracingP6Factory.java

```diff
@@ -113,14 +113,63 @@ public void addsQueryTag() throws Exception {
   }
 
   @Test
-  public void addsRowsWrittenTag() throws Exception {
+  public void addsAffectedRowsTagToPreparedUpdateStatements() throws Exception {
     prepareExecuteUpdate("update t set c='x' where i=2");
 
     assertThat(spans)
       .flatExtracting(s -> s.tags().entrySet())
       .contains(entry("sql.affected_rows", "2"));
   }
 
+  @Test
+  public void addsAffectedRowsTagToPlainUpdateStatements() throws Exception {
+    executeUpdate("update t set c='x' where i=2");
+
+    assertThat(spans)
+      .flatExtracting(s -> s.tags().entrySet())
+      .contains(entry("sql.affected_rows", "2"));
+  }
+
+  @Test
+  public void addsAffectedRowsTagToPlainBatchUpdateStatements() throws Exception {
+    executeBatch("update t set c='x' where i=2", "update t set c='y' where i=1");
+
+    assertThat(spans)
+      .flatExtracting(s -> s.tags().entrySet())
+      .contains(entry("sql.affected_rows", "2,1"));
+  }
+
+  @Test
+  public void doesNotProduceAnySpansForEmptyPlainBatchUpdates() throws Exception {
+    // No SQL at all means no span is started in onBeforeAnyExecute due to there not being any loggable SQL
+    // (see isLoggable)
+    executeBatch();
+
+    assertThat(spans).isEmpty();
+  }
+
+  @Test
+  public void addsAffectedRowsTagToPreparedBatchUpdateStatements() throws Exception {
+    prepareExecuteBatchWithInts("update t set c='x' where i=?", 2, 1);
+
+    assertThat(spans)
+      .flatExtracting(s -> s.tags().entrySet())
+      .contains(entry("sql.affected_rows", "2,1"));
+  }
+
+
+  @Test
+  public void addsEmptyAffectedRowsTagToEmptyPreparedBatchUpdates() throws Exception {
+    // In contrast to the plain statement case, this does produce loggable SQL, so a span is started. Since there are
+    // no entries in the batch, no updates are made, so there are no update counts. This is represented by an empty
+    // string as sql.affected_rows.
+    prepareExecuteBatchWithInts("update t set c='x' where i=?");
+
+    assertThat(spans)
+      .flatExtracting(s -> s.tags().entrySet())
+      .contains(entry("sql.affected_rows", ""));
```

Comment:
We usually avoid zero value tags because they are meaningless and also it is storage cost. I'd rather remove it.

Reply:
OK, fixed in  889e1aa .

Reply:
i see, I wonder what happens when you have three batches and the affected rows are `2,0,3`? Can we cover that with a test?

Reply:
As shown in bfac9c209, that one has "2,0,0", since there are no rows in the test data with i=3. Also added tests for i=0 ("0") and i=2,i=0,i=1 ("2,0,1").